European energy security (#353)
PyPSA/pypsa-eur: doc/release_notes.rst

  * Uniform expansion limit of renewable build-up based on existing capacities can be configured using ``expansion_limit`` option	
    (default: ``false``; limited to determined renewable potentials)	
  * Distribution of country-level capacities proportional to maximum annual energy yield for each bus region	

* The config key ``renewable_capacities_from_OPSD`` is deprecated and was moved under the section, ``estimate_renewable_capacities``. To enable it, set ``from_opsd`` to `True`.	

* Add operational reserve margin constraint analogous to `GenX implementation <https://genxproject.github.io/GenX/dev/core/#Reserves>`_.	
  Can be activated with config setting ``electricity: operational_reserve:``.	

* Add function to add global constraint on use of gas in :mod:`prepare_network`. This can be activated with `electricity: gaslimit:` given in MWh.	

* Add configuration option to implement Energy Availability Factors (EAFs) for conventional generation technologies.	

* A new section ``conventional`` was added to the config file. This section contains configurations for conventional carriers.  

> * Implement country-specific EAFs for nuclear power plants based on IAEA 2018-2020 reported country averages. These are specified under the ``energy_availibility_factor`` key in the config entry ``conventional`` and specify the static `p_max_pu` values.

Reviewer: but i'm not sure if it wouldn't be better to specify the path in the Snakefile instead of the config file. Up to you, I'd say

```suggestion
* Implement country-specific EAFs for nuclear power plants based on IAEA 2018-2020 reported country averages. These are specified in ``data/nuclear-eafs.csv`` and specify the static `p_max_pu` values.
```

Release Notes
##########################################

Upcoming Release
================

* Add an efficiency factor of 88.55% to offshore wind capacity factors
  as a proxy for wake losses. More rigorous modelling is `planned <https://github.com/PyPSA/pypsa-eur/issues/153>`_
  [`#277 <https://github.com/PyPSA/pypsa-eur/pull/277>`_].

* The default deployment density of AC- and DC-connected offshore wind capacity is reduced from 3 MW/sqkm
  to a more conservative estimate of 2 MW/sqkm [`#280 <https://github.com/PyPSA/pypsa-eur/pull/280>`_].

* Following discussion in `#285 <https://github.com/PyPSA/pypsa-eur/issues/285>`_ we have disabled the
  correction factor for solar PV capacity factors by default while satellite data is used.
  A correction factor of 0.854337 is recommended if reanalysis data like ERA5 is used.

> * Resource definitions for memory usage now follow [Snakemake standard resource definition](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#standard-resources) ```mem_mb`` rather than ``mem``.

Author: .rst :)
```suggestion
* Resource definitions for memory usage now follow `Snakemake standard resource definition <https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#standard-resources>`_ ``mem_mb`` rather than ``mem``.
```

    * new: ``estimate_renewable_capacities``	
  * The estimation is endabled by setting the subkey ``enable`` to ``True``.	
  * Configuration of reference year for capacities can be configured (default: ``2020``) 	
  * The list of renewables provided by the OPSD database can be used as a basis, using the tag ``from_opsd: True``. This adds the renewables from the database and fills up the missing capacities with the heuristic distribution. 	
  * Uniform expansion limit of renewable build-up based on existing capacities can be configured using ``expansion_limit`` option	
    (default: ``false``; limited to determined renewable potentials)	
  * Distribution of country-level capacities proportional to maximum annual energy yield for each bus region	

* The config key ``renewable_capacities_from_OPSD`` is deprecated and was moved under the section, ``estimate_renewable_capacities``. To enable it, set ``from_opsd`` to `True`.	

* Add operational reserve margin constraint analogous to `GenX implementation <https://genxproject.github.io/GenX/dev/core/#Reserves>`_.	
  Can be activated with config setting ``electricity: operational_reserve:``.	

> * Add function to add global constraint on use of gas in :mod:`prepare_network`. This can be activated with `electricity: gaslimit:` given in MWh.	

Author: ```suggestion
* Add function to add global constraint on use of gas in :mod:`prepare_network`. This can be activated by including the keyword ``CH4L`` in the ``{opts}`` wildcard which enforces the limit set in ``electricity: gaslimit:`` given in MWh thermal. Alternatively, it is possible to append a number in the `{opts}` wildcard, e.g. `CH4L200` which limits the gas use to 200 TWh thermal.
```

* Add configuration option to implement Energy Availability Factors (EAFs) for conventional generation technologies.	

* A new section ``conventional`` was added to the config file. This section contains configurations for conventional carriers.  

> * Implement country-specific EAFs for nuclear power plants based on IAEA 2018-2020 reported country averages. These are specified `data/nuclear_eafs.csv` and translate to static `p_max_pu` values.

Author: .rst :)
```suggestion
* Implement country-specific EAFs for nuclear power plants based on IAEA 2018-2020 reported country averages. These are specified ``data/nuclear_eafs.csv`` and translate to static ``p_max_pu`` values.
```

* It is now possible to set conventional power plants as extendable by adding them to the list of extendable ``Generator`` carriers in the config.	

* Listing conventional carriers in ``extendable_carriers`` but not in ``conventional_carriers``, sets the corresponding conventional power plants as extendable without a lower capacity bound of today's capacities.	

* Now, conventional carriers have an assigned capital cost by default.	

* The ``build_year`` and ``lifetime`` column are now defined for conventional power plants. 	

* Fix crs bug. Change crs 4236 to 4326.

* Update rasterio version to correctly calculate exclusion raster

* Remove rules to build or retrieve rasterized NATURA 2000 dataset. Renewable potential calculation now directly uses the shapefiles.

* Cache data and cutouts folders. This cache will be updated weekly.

* Add rule to automatically retrieve Natura2000 natural protection areas. Switch of file format to GPKG.

> 

Author: ```suggestion
* Add option to set CO2 emission prices through `{opts}` wildcard: `Ep<number>`, e.g. `Ep180`, will set the EUR/tCO2 price.

* Add option to alter marginal costs of a carrier through `{opts}` wildcard: `<carrier>+m<factor>`, e.g. `gas+m2.5`, will multiply the default marginal cost for gas by factor 2.5.
```
